src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +248,42 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the extension of the file name of this path as a {@code String},
+     * where the extension is defined to be the portion of the file name string
+     * after the last dot ('.').  If the first character in the name is a dot it
+     * is ignored.  If the extension cannot be determined, then the empty string
+     * is returned.  This will occur if the file name has length less than two,
+     * only the first character is a dot, or the last character is a dot.
+     *
+     * @implSpec
+     * If the file name string is the typical case containing at least one dot
+     * at neither the first nor the last character, then the default
+     * implementation is equivalent for this path to:
+     * <pre>{@code
+     *     String filename = getFileName().toString();
+     *     filename.substring(filename.lastIndexOf('.') + 1);
+     * }</pre>
+     *
+     * @return  the extension of the file name of this path, or the empty
+     *          string if the extension is indeterminate
+     */
+    default String getExtension() {
+        String name = getFileName().toString();
+        int length = name.length();
+        String extension = "";
+
+        // Indeterminate if name is too short or equal to "..".
+        if (length > 1 && !name.equals("..")) {
+            int lastDotIndex = name.lastIndexOf('.');
```

Comment:
Since there are no strings of length 0..2, is it sufficient to check for length > 2.
(Instead of the string compare equals("..").

Reply:
Considering that the strings of length 2 are `.x`, `xy`, and `x.` where `x` and `y` are characters which are not `'.'`, then I concur.